src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -68,6 +100,65 @@ internal static partial class ClassifierHelper
             return classifiedSpans;
         }
 
+        public static async Task<ImmutableArray<ImmutableArray<ClassifiedSpan>>> GetNewClassifiedSpansAsync(
+        Document document,
+        ImmutableArray<TextSpan> spans,
+        ClassificationOptions options,
+        bool includeAdditiveSpans,
+        CancellationToken cancellationToken)
+        {
+            var classificationService = document.GetLanguageService<IClassificationService>();
+            if (classificationService == null)
+                return default;
+
+            using var _1 = ArrayBuilder<PooledObject<SegmentedList<ClassifiedSpan>>>.GetInstance(out var syntacticSpansArray);
+            using var _2 = ArrayBuilder<PooledObject<SegmentedList<ClassifiedSpan>>>.GetInstance(out var semanticSpansArray);
+            for (var i = 0; i < spans.Length; i++)
+            {
+                syntacticSpansArray.Add(Classifier.GetPooledList(out _));
+                semanticSpansArray.Add(Classifier.GetPooledList(out _));
+            }
+
+            try
+            {
+                for (var i = 0; i < spans.Length; i++)
+                {
+                    var span = spans[i];
+                    await classificationService.AddSyntacticClassificationsAsync(document, span, syntacticSpansArray[i].Object, cancellationToken).ConfigureAwait(false);
+                    syntacticSpansArray.Add(syntacticSpansArray[i]);
+                }
+
+                await classificationService.AddSemanticClassificationsAsync(document, spans, options, semanticSpansArray, cancellationToken).ConfigureAwait(false);
```

Comment:
>semanticSpansArray [](http://example.com/codeflow?start=102&length=18)

I guess I'm a bit behind on what the plan is for this change, but I would have thought that you could just have this call just put them all in the same array. I assume that wasn't a happy thing to do?